build.gradle

```diff
@@ -14,7 +14,7 @@ buildscript {
         classpath 'org.shipkit:shipkit-auto-version:1.2.2'
 
         classpath 'com.google.googlejavaformat:google-java-format:1.15.0'
-        classpath 'com.android.tools.build:gradle:4.2.0'
+        classpath 'com.android.tools.build:gradle:7.4.0-beta02'
```

Comment:
Reviewer, please make sure you meet the requirements in settings.gradle.kts and import this in your IDE. I'm using the latest stable, theoretically this can be lowered by a few minors without breaking anything.

Reply:
The `beta02` is a bit concerning for release (just being cautious)

Reply:
Google highly encourages using even alpha versions. I think using 7.4.0 beta is better than the previous 7.3.1. Google's "minor" versions are usually major changes with a lot of changes internally. Beta is usually pretty stable already, and the way it's used here (build an empty-ish trivial application) is pretty safe. As soon as the next IntelliJ IDEA release removes the silly arbitrary restriction this can be updated to 7.4.1 stable.

Based on the above we could argue to go down to 7.3.1 and bump to 7.4.1 later when support arrives. Because of the same simplicity.

Note that it is also possible to only include this module if the IDE supports it by adding more conditions to `settings.gradle.kts`, like: `if (System.getProperty("idea.version") ?: "" < "2022.3") { ... }`, but using the latest AGP version is not that necessary.

Reply:
:+1: all set from my side :)

Reply:
I would prefer to use stable versions only, so let's go with `7.3.1` for now. When `7.4.0` has been published, we can indeed update our toolchain.

Reply:
7.4.1 is already stable and it would pass CI, but no-one using IDEA would be able to open the project. I'll try 7.3.1 for now then.

Reply:
@TimvdLippe 7.3.1 (downgraded in https://github.com/mockito/mockito/pull/2894/commits/79d032f0255215af3f45f22c2867f0ec310959b9) looks OK apart from Android Lint being too old for Kotlin 1.8+:
```
> Task :androidTest:lintAnalyzeDebug
e: P:/caches/gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.8.10/6d5560a229477df9406943d5feda5618e98eb64c/kotlin-stdlib-1.8.10.jar!/META-INF/kotlin-stdlib.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.
e: P:/caches/gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.8.10/6d5560a229477df9406943d5feda5618e98eb64c/kotlin-stdlib-1.8.10.jar!/META-INF/kotlin-stdlib-jdk8.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.
e: P:/caches/gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.8.10/6d5560a229477df9406943d5feda5618e98eb64c/kotlin-stdlib-1.8.10.jar!/META-INF/kotlin-stdlib-jdk7.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.
e: P:/caches/gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-common/1.8.10/a61b182458550492c12aee66157d7b524a63a5ec/kotlin-stdlib-common-1.8.10.jar!/META-INF/kotlin-stdlib-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.
e: P:/projects/contrib/github-mockito/subprojects/androidTest/build/intermediates/compile_app_classes_jar/debug/classes.jar!/META-INF/androidTest_debug.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.
```
but this is an acceptable temporary loss on this project.